### ISSUE TITLE
Support network visulazation that is not relaterd to a know framework

### ISSUE THREAD

**Issue by yoelp2k** [lutzroeder/netron#688]
Support visualization of neural networks tools that are not necessarily related to any framework   
- A loose json format that allow defining any layer
- tensor shape, data type for all inputs/outputs of layers
- user defined properties for any layer 
- fusion of layers 

It seems that these features are already supported by netron. 
The main issue is describing the graph input data structure and stitching it all together 

This feature will be most helpful for anyone developing neural networks that are not supported buy netron without the need to write specific reader

**Comment by lutzroeder** [comment]
@yoelp2k what is the scenario that would require this for a broad set of user?

**Comment by yoelp2k** [comment]
Hi,

First of all thank you for developing netron! it is a wonderful tool that
really helps us, and thank you for following up on this issue

In short: We need some way to visualize a platform independent network. We
develop a tool that does some network agnostic operations and interested in
looking at the intermediate stages. and I need to produce the serialization
file from several languages (scala/java/python)

Here is to longer version
I work at Sony Israel Semiconductor. We are developing a dnn inference chip
for inference of networks from various frameworks.
In order to load a network into the chip we need to prepare it. We convert
the input network (tensorflow/pytorch/caffe2 etc) to our own network
agnostic format, explore/analyze and fix before returning to the framework
again
These steps are much harder to do on the frameworks models. Also they are
common to all frameworks - so it makes much sense converting to our
internal format
Specifically, we are developing a layer quantization aid tool to add
quantization layers with minimal loss

I'm interested in exporting intermediate networks from multiple languages:
scala/java and python

First I tried exporting our network in MXNet json format but found no
documentation of this format so the result is pretty lame.
I also looked at ONNX scala serialization plugin but it seems hard wired to
the whole framework. But I admit that was an impression made by looking
briefly at some APIs and I haven't put any real effort using it though

I'm not interested in any training/inference capabilities - just skeletal
representation of the network.
I'm interested in visualizing custom layers, displaying different
attributes per-layer (quantization info for instance). Also fusing several
layers (conv + activations for instance)

Perhaps you can recommend someplace I can read more about MXNet
serialization so I can produce better results?
Or perhaps you can recommend some other framework serialization that will
allow me to define custom fields and custom layers?

Thank you very much in advance,
Yoel.

**Comment by prasanthpul** [comment]
@yoelp2k have you considered using ONNX as your framework agnostic format? ONNX's purpose is to serve as a common format that represents networks from many different frameworks. It also supports custom fields and custom layers. The format is documented at https://github.com/onnx/onnx/blob/master/docs/IR.md and there are converters for all the major frameworks

**Comment by yoelp2k** [comment]
@prasanthpul Thank you. I will look into it.
